### PR TITLE
#164198185: delivers read, update, delete articles

### PR DESCRIPTION
**Pull request by thislekan**
#### What does this PR do?

- provide route for get all articles
- provide route for get single article
- provide route for update single article
- provide route for delete single article

#### Description of Task to be completed?

- [x] create a route for ```GET single article``` 
- [x] create a route for ```GET all articles```
- [x] create a route for ```DELETE single article```
- [x] create a route for ```UPDATE single article```

- [x] restricts update of articles to only article's author
- [x] restricts delete of articles to only article's author

#### How should this be manually tested?

- clone/download this repository
- install node on your system if you haven't
- change directory into the now downloaded/clone folder
- create a ```.env``` file from the available ```.env.examples``` file
- create a ```.env.test``` file with the same procedure as above
- change the ```DB_DATABASE``` value in the ```.env.test``` like so:

```
DB_DATABASE=anythingyouwishtocallit
```

- run npm install
- run npm run migrate
- create an article from the user route ```/api/v1/articles/```
- get all articles from the get all articles route ```/api/v1/articles/``` with a ```GET``` method
- get single article from the get single article route ```/api/v1/articles/:slug``` with a ```GET``` method
- update single article from the update single article route ```/api/v1/articles/:id``` with a ```PUT``` method
- delete single article from the delete single article route ```/api/v1/articles/:id``` with a ```DELETE``` method

#### Any background context you want to provide?

- The above procedure assumes a little familiarity with programming.
- The above procedure absolutely requires creation of a user

#### What are the relevant pivotal tracker stories?

[#164198185](https://www.pivotaltracker.com/story/show/164198185)

### CODE REVIEW

File: controllers/articlesController.js [hunk 1]
```diff
@@ -27,9 +27,127 @@ export default class ArticlesController {
...
+        res, STATUS.CREATED, article.dataValues, 'article was successfully created', true,
+      );
     } catch (error) {
-      return Response.send(res, STATUS.BAD_REQUEST, error, 'failed');
+      return Response.send(res, STATUS.BAD_REQUEST, error, false);
+    }
+  }
+
+  /**
+   * Sends the request to the database and returns the article object
```
Comment (aanchirinah) [inline controllers/articlesController.js]: #### Nitpick
- Not descriptive of what actually goes on in this function.

File: controllers/articlesController.js [hunk 2]
```diff
@@ -27,9 +27,127 @@ export default class ArticlesController {
...
+      );
+    } catch (error) {
+      return Response.send(
+        res, STATUS.BAD_REQUEST, error, 'server error', false,
+      );
+    }
+  }
+
+  /**
+ * Sends the articleId parameter to the database and returns the corresponding article object
```
Comment (aanchirinah) [inline controllers/articlesController.js]: #### Nitpick
- Really? I thought it rather updates an article object?

File: controllers/articlesController.js [hunk 3]
```diff
@@ -27,9 +27,127 @@ export default class ArticlesController {
...
+      return Response.send(
+        res, STATUS.OK, article[1], 'article was successfully updated', true,
+      );
+    } catch (error) {
+      return Response.send(res, STATUS.BAD_REQUEST, error, false);
+    }
+  }
+
+  /**
+ * Sends the articleId parameter to the database and returns the corresponding article object
```
Comment (aanchirinah) [inline controllers/articlesController.js]: #### Nitpick
- Really? I thought it deletes an article object?
- Also, did we decide on archiving an article or deleting it?

File: middlewares/articlesMiddleware.js [hunk 4]
```diff
@@ -29,37 +30,163 @@ export default class AriclesMiddleware {
...
+    if (typeof slug !== 'string') {
+      return Response.send(
+        res, STATUS.BAD_REQUEST, [], 'slug is not a string', false,
+      );
+    }
+    return next();
+  }
+
+  /**
+   * Sends the articleId parameter to the database and returns the corresponding article object
```
Comment (aanchirinah) [inline middlewares/articlesMiddleware.js]: #### Nitpick
- Make your JSDOCs be much more descriptive.

File: test/unit/middlewares/articlesMiddleware.test.js [hunk 5]
```diff
@@ -262,4 +299,159 @@ describe('API endpoint: /api/articles (Middleware test)', () => {
...
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.haveOwnProperty('code').to.equal(STATUS.BAD_REQUEST);
+          expect(res.body.message).to.be.equal('article Id is not valid');
+          done();
+        });
+    });
+  });
+
+  describe('PUT: /api/v1/articles/:articleId', () => {
+    it('Should return an error if authorId is incorrect', (done) => {
```
Comment (aanchirinah) [inline test/unit/middlewares/articlesMiddleware.test.js]: #### Nitpick
- AuthorId or ArticleId ?

**Comment by aanchirinah** [comment]
@thislekan , You have merge conflicts also.

File: controllers/articlesController.js (hunk 3 above)
Comment (thislekan) [reply controllers/articlesController.js]: We decided on archiving articles when the user wants to delete their accounts. Nothing decided about articles yet.

File: test/unit/middlewares/articlesMiddleware.test.js (hunk 5 above)
Comment (thislekan) [reply test/unit/middlewares/articlesMiddleware.test.js]: I'm sorry for all these details errors. Stress and pressure is the cause. I'll attend to them all

**Pull request merged**